Cargo.toml

```diff
@@ -10,5 +10,3 @@ rpath = true
 [profile.release]
 rpath = true
 
-[profile.test]
-opt-level = 2
```

Comment:
I'd propose a two-step process. We can create and use a separate profile for CI in which we can set the opt-level to 0.

Re point 2: I think we should validate locally whether the compilation + run time is also lower when using `opt-level = 0`

Moreover, we should validate how long the [`test-recent-mainnet-block`](https://github.com/NomicFoundation/edr/blob/main/.github/workflows/test-recent-mainnet-block.yml) workflow takes with opt-levels 0, 1, and 2 such that we ensure that that's running as fast as possible in CI too:

Reply:
Ah, good catch with `test-recent-mainnet-block`! I didn't realize we have tests where this might have an impact; will test this next. Are there any other tests or workflows that may be sensitive to that?

Also, do we expect to often locally run expensive tests in general? Is it useful? Sorry if that's a dumb question, I'm still trying to figure out what's what in EDR 😅

Reply:
The ones in the pre-commit hook are the tests that are part of the normal workflow. Whichever optimisation level takes least time there, would be preferable. That's the only thing you'd need to consider for that decision, I think :)

Reply:
Posted local results in https://github.com/NomicFoundation/edr/pull/501#issuecomment-2158016805

Reply:
FWIW I'm pretty sure that the `test-recent-mainnet-block` (execution) times are _completely_ dominated by networking (and by the very structure of the tests, these are never cahced), so I would be surprised if the optimization level makes a difference.